### PR TITLE
Draft of image patch extraction pipeline - for feedback

### PR DESCRIPTION
**Pull request by tim-rehnstrom**
This is a draft containing the patch extraction pipeline. Please let me know if you find any errors or if you see something where I can optimize the code.

### CODE REVIEW

File: .devcontainer/analyst/Dockerfile [hunk 1]
```diff
@@ -1,2 +1,3 @@
 FROM ghcr.io/nasa/isaac_analyst_notebook
 ENV DEBIAN_FRONTEND=dialog
+RUN pip install torch==2.2.1 torchvision==0.17.1 torchaudio==2.2.1 --index-url https://download.pytorch.org/whl/cu121
```
Comment (bcoltin) [inline .devcontainer/analyst/Dockerfile:3]: Do we need to commit to these exact versions? What is this url?

File: .devcontainer/analyst/devcontainer.json [hunk 2]
```diff
@@ -11,37 +11,45 @@
...
-		"twxs.cmake",
-		"yzhang.markdown-all-in-one",
-		"zachflower.uncrustify"
-	],
+	
 	"mounts": [
 		"source=${localWorkspaceFolder},target=/src/isaac/src,type=bind,consistency=cached",
-		"source=${localWorkspaceFolder}/../../astrobee/src,target=/src/astrobee/src,type=bind,consistency=cached"
+		"source=${localWorkspaceFolder}/../../astrobee/src,target=/src/astrobee/src,type=bind,consistency=cached",
+		"source=${localWorkspaceFolder}/../../data/Sock_example,target=/src/data/vent,type=bind,consistency=cached"
```
Comment (bcoltin) [inline .devcontainer/analyst/devcontainer.json:52]: How does this directory get here? Will this cause problems for people who don't have your exact setup or use the analyst notebook for other purposes?

File: analyst/workspace/.Trash-0/files/extract_target.ipynb
Comment (bcoltin) [inline analyst/workspace/.Trash-0/files/extract_target.ipynb:1]: Added by mistake?

File: analyst/workspace/1_import_bagfiles.ipynb [hunk 3]
```diff
@@ -23,16 +23,99 @@
...
+      "Topics found in data/bags/2024-03-21_tim/bsharp/Fixed/20240321_2254_survey_bsharp7_0.fix_all.bag:\n",
+      "{'/hw/depth_haz/points': 297, '/hw/depth_haz/extended/amplitude_int': 297, '/gnc/ekf': 5381, '/hw/imu': 6996, '/loc/pose': 5363, '/loc/of/features': 488, '/graph_loc/state': 503, '/sparse_mapping/pose': 85, '/mob/flight_mode': 1, '/hw/wifi': 606, '/mgt/img_sampler/nav_cam/image_record': 255, '/hw/cam_nav/image_color': 255, '/hw/cam_nav_bayer': 255, '/hw/cam_nav_ctrl': 61, '/hw/depth_haz/extended': 297, '/beh/inspection/status': 303, '/hw/cam_sci_info': 121, '/tf': 5379, '/loc/ml/features': 84, '/mgt/sys_monitor/time_sync': 120}\n",
+      "Done loading database\n"
+     ]
+    }
+   ],
    "source": [
     "from scripts.load_bag_database import LoadBagDatabase\n",
-    "path=\"data/bags/\"\n",
+    "path=\"data/bags/2024-03-21_tim/bsharp/Fixed\"\n",
```
Comment (bcoltin) [inline analyst/workspace/1_import_bagfiles.ipynb:115]: not sure you want to commit the changes to this file. They seem specific to you.

File: analyst/workspace/extract_query_target.ipynb
Comment (bcoltin) [inline analyst/workspace/extract_query_target.ipynb:1]: Fix spelling errors

Comment (bcoltin) [reply analyst/workspace/extract_query_target.ipynb:1]: How do other people get the data to run this?

File: analyst/workspace/scripts/query_image_of_bag.py
Comment (bcoltin) [inline analyst/workspace/scripts/query_image_of_bag.py:1]: Make this a function query_image.py? Probably doesn't need a separate module?

File: analyst/workspace/scripts/save_patch_scripts.py
Comment (bcoltin) [inline analyst/workspace/scripts/save_patch_scripts.py:1]: Probably better not to have _scripts in filename

File: analyst/workspace/target_select_workspace.ipynb
Comment (bcoltin) [inline analyst/workspace/target_select_workspace.ipynb:1]: What is this? Add some explanation

File: analyst/workspace/testing_of_warp.ipynb
Comment (bcoltin) [inline analyst/workspace/testing_of_warp.ipynb:1]: what is this? add comments

**Review by bcoltin — CHANGES_REQUESTED** [review]
My main suggestions are to make sure it is easy to run everything on other people's computers; and where it makes sense, try to put things in the jupyter notebooks into reusable functions in libraries.

File: .devcontainer/analyst/Dockerfile (hunk 1 above)
Comment (marinagmoreira) [reply .devcontainer/analyst/Dockerfile:3]: the link is from the pytorch install instructions here: https://pytorch.org/

Comment (marinagmoreira) [reply .devcontainer/analyst/Dockerfile:3]: the exact versions might be remaining from when we had to support multiple ubuntu versions

**Comment by marinagmoreira** [comment]
I recommend that you work from a branch with a relevant name to your feature and not develop. This makes it easier for us to checkout your branch and fix something that comes up too.

To create a branch you can do:
`git checkout -b switch_id`
to push the branch to your origin:
`git push -u origin switch_id`
and then re-open a new PR

Also, keep your branch updated with our develop, since we push fixes and updates frequently. Right now, your tests in this PR are failing because you haven't merged in the latest opencv updates

**Pull request closed without merging**